freqtrade/freqai/data_kitchen.py

```diff
@@ -843,14 +950,14 @@ def create_fulltimerange(self, backtest_tr: str, backtest_period_days: int) -> s
                                        "Please indicate the end date of your desired backtesting. "
                                        "timerange.")
             # backtest_timerange.stopts = int(
-            #     datetime.datetime.now(tz=datetime.timezone.utc).timestamp()
+            #     datetime.now(tz=timezone.utc).timestamp()
             # )
 
         backtest_timerange.startts = (
             backtest_timerange.startts - backtest_period_days * SECONDS_IN_DAY
         )
-        start = datetime.datetime.utcfromtimestamp(backtest_timerange.startts)
-        stop = datetime.datetime.utcfromtimestamp(backtest_timerange.stopts)
+        start = datetime.utcfromtimestamp(backtest_timerange.startts)
+        stop = datetime.utcfromtimestamp(backtest_timerange.stopts)
```

Comment:
This is ... wrong (sorry for not noticing this before - big reviews ... :unamused: ) - and as such dangerous (even though it'll probably not have a big impact in this location specifically).

using `utcfromtimestamp()` results in a "broken" datetime object - unless your system is in the UTC timezone.
while the time will be in utc - the object will be tz native (in your system timezone).
as a consequence - running `datetime.utcfromtimestamp(backtest_timerange.startts).timestamp()` will return a timestamp shifted by your timezone offset.

The correct way for this is the below:
`datetime.fromtimestamp(timerange.startts, tz=timezone.utc)`

obviously applies to all usages of `utcfromtimestamp()` ...

Reply:
Thanks for noticing, I will fix this and any other occurrences.